### ISSUE TITLE
Math size not accounted for when resolving `em` units

### ISSUE THREAD

**Issue by EpicEricEE** [typst/typst#5164]
### Description

When using `em` units in math mode together with script sized content, the resolved length is not adjusted to the scaled font size.
For example:
<table>

<tr>
<th>With <code>1em</code> spacing</th>
<th>With unicode 'Em Space'</th>
</tr>

<tr>

<td>

```typ
$ a #h(1em) b \
  x^(a #h(1em) b) \
  x^x^(a #h(1em) b) $
```

</td>

<td>

```typ
$ a\u{2003}b \
  x^(a\u{2003}b) \
  x^x^(a\u{2003}b) $
```

</td>
</tr>

<tr>
<td>

![image](https://github.com/user-attachments/assets/e15fcc2d-74c9-48c6-934e-eec32742d2fd)

</td>

<td>

![image](https://github.com/user-attachments/assets/a8b32b97-c26b-4e9b-bf38-29be1dce9955)

</td>
</tr>

</table>

This also manifests itself when setting the size of an accent, as seen in [How to define uniformly larger accents in math? (Forum)](https://forum.typst.app/t/how-to-define-uniformly-larger-accents-in-math/983)

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
I think it’s the same as https://github.com/typst/typst/issues/3729

**Comment by Enivex** [comment]
> I think it’s the same as #3729

The accent scaling in scripts from the forum post makes this a less academic issue than I initially thought based on just the title.

**Comment by EpicEricEE** [comment]
I was looking into this a bit, and I think we can fix this independent from #3729.

In many math elements, font size dependent lengths are already resolved via
```rs
let font_size = scaled_font_size(ctx, styles)
let short_fall = DELIM_SHORT_FALL.at(font_size);
```
instead of just `DELIM_SHORT_FALL.resolve(styles)` which uses the unscaled font size of the base equation.

Can we not do the same with the remaining calls to `resolve` that exist in the layout of  `accent`, `cancel`, `stretch` and horizontal spacings? The only missing thing then would be fields marked with `#[resolve]` (mostly strokes and gaps) as they automatically run through the "wrong" function.

**Comment by laurmaedje** [comment]
Could work. Feel free to try.
